view/frontend/web/js/product/button.js

```diff
@@ -0,0 +1,52 @@
+define([
+    'jquery',
+    'underscore'
+],function ($, _) {
+
+    return function (config) {
+
+        $(config.buttonId).on('mouseover',function (e) {
+
+            var radioAndCheckbox = {};
+            var inputElements = [];
+
+            $.each($('.product-options-wrapper [aria-required="true"]'),function () {
+
+                if(typeof $(this).prop("name") === "undefined" || $(this).attr("type") === "hidden"){
+                    return;
+                }
+
+                if($(this).attr("type") === "radio" || $(this).attr("type") === "checkbox"){
+
+                    String($(this).attr("name")) in radioAndCheckbox ?
+                        radioAndCheckbox[$(this).attr("name")].push($(this).is(':checked'))
+                        : radioAndCheckbox[$(this).attr("name")] = [$(this).is(':checked')]
+                    ;
+
+                } else {
+
+                    inputElements.push(Boolean($(this).val()));
+                }
+            });
+
+            inputElements = _.every(inputElements);
```

Comment:
I think using the same variable name, but going from an array to a boolean value is not ideal, could you change the variable being set to something more boolean-like, maybe like 'areInputElementsSet' or something similar